### ISSUE TITLE
Spyder freezes on quit

### ISSUE THREAD

**Issue by giuseppedelnapalle** [spyder-ide/spyder#19814]
### Issue Report Checklist

* [y ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [y ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [y ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [y ] Restarted Spyder
    * [y ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [y ] Tried the other applicable steps from the Troubleshooting Guide
* [y ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
Spyder always freezes when attempting to quit. On rare occasions it quits successfully.

### What steps reproduce the problem?

1. launch Spyder from the terminal
2. lick File/Quit or the x button at the upper right corner
3. 

### What is the expected output? What do you see instead?
Spyder quits smoothly.
Spyder freezes (not responding to any actions). 

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 5.3.3  (conda)
* Python version: 3.9.12 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Linux 5.19.14-1-MANJARO

### Dependencies

```# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<8.0.0       :  7.31.1 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.16.0 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.4.4 (OK)
numpydoc >=0.6.0              :  1.4.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0;<3.0           :  2.14.5 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.5.0;<1.6.0          :  1.5.0 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.3.2;<5.4.0      :  5.3.2 (OK)
qtpy >=2.1.0                  :  2.2.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  63.4.1 (OK)
sphinx >=0.6.6                :  5.0.2 (OK)
spyder_kernels >=2.3.3;<2.4.0 :  2.3.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
xdg >=0.26                    :  0.27 (OK)
zmq >=22.1.0                  :  23.2.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=3.0.0            :  None (NOK)
numpy >=1.7                   :  None (NOK)
pandas >=1.1.1                :  None (NOK)
scipy >=0.17.0                :  None (NOK)
sympy >=0.7.3                 :  None (NOK)

```

**Comment by ccordoba12** [comment]
Hey @giuseppedelnapalle, thanks for reporting. I have a question for you: are you using Wayland? I say it because that could be the cause of this issue.

**Comment by giuseppedelnapalle** [comment]
Thank you for the prompt response. I checked the software manager and found that wayland 1.21 was installed as a dependency (not installed explicitly). It is required by many packages, including egl-wayland, gst-plugins-bad, gtk3, gtk4, intel-media-sdk, lib32-wayland, libva, mesa, vulkan-intel, vulkan-redeon, webkit2gtk, webkit2gtk-4.1, wpebackend-fdo.

btw, I did stumble upon another issue related to libstdc++.so.6 when launching Spyder from terminal. I found a workaround following [this post](https://stackoverflow.com/questions/72110384/libgl-error-mesa-loader-failed-to-open-iris/72427700#72427700). idk if this is relevant to the freezing issue. Feel free to ask for more info to find the cause.

**Comment by ccordoba12** [comment]
@giuseppedelnapalle, to check if this related to Wayland, please logout from your session, and log in but to Xorg. I don't know what desktop environment you're using, but you should see an option like the one displayed here in your login screen:

https://docs.fedoraproject.org/en-US/quick-docs/configuring-xorg-as-default-gnome-session/#proc-configuring-xorg-as-default-gnome-session

So you need to select in and then proceed to log in.

**Comment by giuseppedelnapalle** [comment]
Thank you for the reply. As far as I can tell, I am using Xorg. I checked the XDG_SESSION_TYPE variable and the value was x11. So it is not related to Wayland. Sometimes Spyder quits successfully but I can't see what action is taken to make it happen.

**Comment by ccordoba12** [comment]
Ok, then please open a terminal and run there

    spyder --debug-info verbose --debug-output terminal

After Spyder starts, please wait a minute or so and then close it. Afterwards, please copy the contents of your terminal and post them here. Perhaps that will show us where Spyder is freezing.

**Comment by giuseppedelnapalle** [comment]
The output was uploaded as a text file as the comment exceeded the maximum characters. Please check it out.
[debug_spyder.txt](https://github.com/spyder-ide/spyder/files/9843704/debug_spyder.txt)

**Comment by ccordoba12** [comment]
There are some errors at the beginning of that file that could be causing the freeze on exit:

```
libGL error: MESA-LOADER: failed to open iris: /home/mario/miniconda3/lib/python3.10/site-packages/zmq/backend/cython/../../../../.././libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/lib/dri/iris_dri.so) (search paths /usr/lib/dri, suffix _dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open iris: /home/mario/miniconda3/lib/python3.10/site-packages/zmq/backend/cython/../../../../.././libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/lib/dri/iris_dri.so) (search paths /usr/lib/dri, suffix _dri)
libGL error: failed to load driver: iris
libGL error: MESA-LOADER: failed to open swrast: /home/mario/miniconda3/lib/python3.10/site-packages/zmq/backend/cython/../../../../.././libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /usr/lib/dri/swrast_dri.so) (search paths /usr/lib/dri, suffix _dri)
libGL error: failed to load driver: swrast
```

They are the only errors present and could be related to your graphics card.

**Comment by ccordoba12** [comment]
Another question for you: did you install Spyder using Conda-forge or the defaults channel?

**Comment by giuseppedelnapalle** [comment]
The defaults channel. Actually the error at the beginning can be partly fixed (not reporting the error) by copying /usr/lib/x86_64-linux-gnu/libstdc++.so.6 to /home/$USER/miniconda/lib, but the exit error still exists. If it's indeed the culprit, do you have any suggestions?

**Comment by ccordoba12** [comment]
Please create a new environment with Conda-forge packages using the following commands

    conda create -n spyder-env -c conda-forge spyder
    conda activate spyder-env
    spyder

and try again. Perhaps that solves the problem.

**Comment by ccordoba12** [comment]
> Thanks for the suggestion. It seems the issue was fixed, or at least partly fixed

That's great news! I guess the main issue is that Qt (the graphical toolkit on which Spyder is based) is not packaged correctly in Anaconda, but it is in Conda-forge.

> Spyder is closed successfully, but it takes quite a long time (10-40s). It seems the time depends on how long Spyder is on before closing it. Longer latency is associated with quicker exit. Is this a minor bug or a issue specific to the hardware and the system?

Looking at your logs (specifically at the timestamps), it seems the latency issue happens when deleting the Application plugin:

```
2022-10-24 19:29:56,830 [DEBUG] [spyder.api.plugin_registration.registry] -> Deleting plugin application
2022-10-24 19:30:20,180 [DEBUG] [spyder.api.plugin_registration.registry] -> Deleting plugin appearance
```

My guess is that this is related to us checking for Spyder updates at startup, which runs in a thread, takes a long time and can't be interrupted if in progress when closing Spyder. That should be solved in our next version because we delayed starting that thread for a minute after Spyder starts, considering that at that point users should be invested on having a long coding session.

You should be able to avoid that problem now by going to

    Tools > Preferences > Application > Advanced settings

and disabling the option `Check for updates on startup`.

Let us know if that works for you.

**Comment by giuseppedelnapalle** [comment]
Spot on! Thank you for your help. Hope this annoying issue will be fixed in the upcoming version.

**Comment by ccordoba12** [comment]
Glad it worked for you @giuseppedelnapalle! Thanks for letting us know about it.

**Comment by NotNormallyAGitUser** [comment]
I had similar behaviour with Spyder 5.5.1.  I issued "conda update" for ipykernel and Spyder (the latter was already up-to-date) as well as resetting all Spyder settings.  I then saw a handful of Spyder processes in the Task Manager (Windows 10) and dozens of Python processes, many of which resisted "End Task".  In the end, I shut down all apps and rebooted.  This allowed Spyder to quit, and no zombie processes showed up in the Task Manager under Spyder or Python.  I did, however, have to re-do customizations, e.g., re-specify startup.py.
